### ISSUE TITLE
Clarify the source of postData variable in the guide

### ISSUE THREAD

**Issue by niamleeson** [emberjs/guides#1738]
https://guides.emberjs.com/v2.9.0/components/wrapping-content-in-a-component/

In this block of code, it is not clear where the variable `postData` in `postData=postData` is coming from.
```javascript
// app/templates/components/blog-post.hbs
<h2>{{title}}</h2>
<div class="body">{{yield (hash body=(component editStyle postData=postData))}}</div>
```

Could you expand the context a little more?

Thank you.

**Comment by toddjordan** [comment]
The component is showing a blog post, but yielding its data to its block so that the consumer of the component has control over how to format it. 

`postData` is essentially the text from the blog post.  Its showing that you can pass arguments into the component helper so that the consumer of the component (in this case index.hbs) doesn't have to worry about passing data.

Some text on what postData is should do, and maybe renaming it to something like `blogText`

**Comment by niamleeson** [comment]
Is it something like this?
```javascript
// app/index.js
import Ember from 'ember';

export default Ember.Component.extend({
  blogTexts: [..., ..., ...]
});

// app/templates/index.hbs
{{#each blogTexts as |blogText|}}
  {{#blog-post editStyle="markdown-style" postData=blogText as |post|}}
    <p class="author">by {{author}}</p>
    {{post.body editStyle="compact-style"}}
  {{/blog-post}}
{{/each}}
```

Or more like this?
```javascript
// app/templates/index.hbs
{{#blog-post editStyle="markdown-style" as |post|}}
  <p class="author">by {{author}}</p>
  {{post.body}}
{{/blog-post}}

// app/components/blog-post.js
import Ember from 'ember';

export default Ember.Component.extend({
  blogText: '...'
});

// app/templates/components/blog-post.hbs
<h2>{{title}}</h2>
<div class="body">{{yield (hash body=(component editStyle postData=blogText))}}</div>
```

Thank you.

**Comment by toddjordan** [comment]
A blog post would have one text, so the second.  It might be simpler to bind the text down like this:

```javascript
// app/templates/index.hbs
{{#blog-post editStyle="markdown-style" text=myText as |post|}}
  <p class="author">by {{author}}</p>
  {{post.body}}
{{/blog-post}}

// app/templates/components/blog-post.hbs
<h2>{{title}}</h2>
<div class="body">{{yield (hash body=(component editStyle text=text))}}</div>
```

**Comment by niamleeson** [comment]
I see. I've submitted a PR on this issue. I think the newcomers will benefit from a little more detailed explanation on where the variables are coming from. Please take a look and let me know what you think.

**Comment by toddjordan** [comment]
Addressed by #1740 ❤️
